packages/php-config-printer/src/NodeTraverser/ImportFullyQualifiedNamesNodeTraverser.php

```diff
@@ -27,70 +39,72 @@ public function __construct(
      */
     public function traverseNodes(array $nodes): array
     {
-        $nameImports = $this->collectNameImportsFromNodes($nodes);
-        if ($nameImports === []) {
-            return $nodes;
-        }
+        $this->collectNameImportsFromNodes($nodes);
+
+        $imports = array_unique($this->importFullyQualifiedNamesNodeVisitor->getImports());
 
-        return $this->addUseImportsToNamespace($nodes, $nameImports);
+        return $this->addUseImportsToNamespace($nodes, $imports);
     }
 
     /**
      * @param Node[] $nodes
-     * @param string[] $nameImports
+     * @param FullyQualifiedImport[] $imports
      * @return Node[]
      */
-    private function addUseImportsToNamespace(array $nodes, array $nameImports): array
+    private function addUseImportsToNamespace(array $nodes, array $imports): array
     {
-        if ($nameImports === []) {
+        if ($imports === []) {
             return $nodes;
         }
 
-        sort($nameImports);
+        usort($imports, [$this, 'sortImportsByFullyQualified']);
+        usort($imports, [$this, 'sortImportsByType']);
```

Comment:
Could you extract this logic to a new sorting service? It's quite a big piece of logic now. I think we'll use it somewhere else too.

Ideally without array method magic (I've realized we miss PHPStan rule for that :))

Reply:
Done!